lib/routes/_core_profile.js

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Joi = require('joi');
+
+const AppError = require('../error');
+const config = require('../config');
+const logger = require('../logging')('routes._core_profile');
+const request = require('../request');
+
+const AUTH_SERVER_URL = config.get('authServer.url') + '/account/profile';
+
+/**
+ * This is an internal-use route that retreives all the user
+ * profile data available from the auth-server.  Various public
+ * routes can call into it and return a subset of the info.
+ */
+module.exports = {
+  isInternal: true,
+  auth: {
+    strategy: 'oauth',
+    scope: ['profile', 'profile:email', 'profile:locale', 'profile:amr', /* openid-connect scope */'email' ]
+  },
+  response: {
+    schema: {
+      email: Joi.string().optional(),
+      locale: Joi.string().optional(),
+      amrValues: Joi.array().items(Joi.string().required()).optional(),
+      twoFactorAuthentication: Joi.boolean().optional()
+    }
+  },
+  handler: function _core_profile(req, reply) {
```

Comment:
The logic here is coped pretty directly from the old `/email` route, including error-handling etc.